### PR TITLE
Abstract validation - refactoring

### PR DESCRIPTION
**Pull request by pqienso**
The `Validator` classes' only responsibility now is to check if all required fields have some input. Its only public method, `.validateCommand()`, returns `void` and throws `Exceptions.InvalidInput` if there are missing required arguments or too many arguments. 
`validateCommand()` is only called once in each `Manager` class at the start of its `.execute()` method. Thereafter, we can be assured that all required fields have some form of user input, and that there are no unnecessary inputs by the user.
**NOTE:**
_The validity of the data in each required field_ (e.g. whether the input is a valid integer / is a valid date) is verified by the `Manager` classes themselves.
`HelpMenuManager` has also been refactored and cleaned up. Test codes have been changed accordingly.

### CODE REVIEW

File: src/main/java/byteceps/ui/strings/ManagerStrings.java [hunk 1]
```diff
@@ -30,11 +30,20 @@ public class ManagerStrings {
...
     public static final String WORKOUTS = "Workouts";
-    public static final String INCOMPLETE_INFO = "info command not complete. " +
+    public static final String INCOMPLETE_INFO = "Info command not complete. " +
             "Please use: workout /info <WORKOUT_PLAN_NAME [string]> ";
+    public static final String INCOMPLETE_CREATE = "Create command not complete. " +
+            "Please use: workout /create <NEW_WORKOUT_PLAN_NAME [string]>";
+    public static final String INCOMPLETE_DELETE_WORKOUT = "Delete command not complete." +
+            "Please use: workout /delete <WORKOUT_PLAN_NAME [string]>";
+    public static final String INCOMPLETE_SEARCH = "Search term cannot be left blank." +
+            "Please use : workout /search <WORKOUT_PLAN_NAME [string]>";
```
Comment (joshualeejunyi) [inline src/main/java/byteceps/ui/strings/ManagerStrings.java:46]: Might want to standardize if there's a space before the colon or not (also in line 34)

File: data.json.old_20240407_204039 [hunk 2]
```diff
@@ -0,0 +1 @@
+{"workoutManager":[{"exerciseList":[],"activityName":"w","exerciseSet":[]}],"weeklyProgram":{"WEDNESDAY":"","MONDAY":"","THURSDAY":"","SUNDAY":"w","TUESDAY":"","FRIDAY":"","SATURDAY":""},"exerciseManager":[{"activityName":"e"}],"WorkoutLogManager":[{"exercises":[{"exerciseName":"e","reps":-1,"sets":-1,"weight":-1}],"workoutDate":"2024-04-07","workoutName":"w"}]}
```
Comment (joshualeejunyi) [inline data.json.old_20240407_204039:1]: we should add the old jsons to .gitignore LOL

File: src/main/java/byteceps/activities/Day.java [hunk 3]
```diff
@@ -10,6 +10,7 @@ public void setAssignedWorkout(Workout assignedWorkout) {
         this.assignedWorkout = assignedWorkout;
     }
 
+
```
Comment (joshualeejunyi) [inline src/main/java/byteceps/activities/Day.java:13]: Shall we try to keep newlines to a minimum haha

**Review by joshualeejunyi — COMMENTED** [review]
Look fine thanks!

**Comment by V4Vern** [comment]
LGTM

**Pull request merged**